libs/dh/admin/feature-user-management/src/lib/details/details.component.ts

```diff
@@ -17,144 +17,122 @@
   changeDetection: ChangeDetectionStrategy.OnPush,
-  selector: 'dh-user-drawer',
+  selector: 'dh-user-details',
   standalone: true,
   templateUrl: './details.component.html',
   imports: [
-    TranslocoDirective,
+    RouterOutlet,
     MatMenuModule,
+    TranslocoDirective,
 
     WATT_TABS,
     WATT_MODAL,
     WATT_DRAWER,
     WattButtonComponent,
 
+    DhUserRolesComponent,
+    DhDeactivteComponent,
+    DhReactivateComponent,
     DhUserStatusComponent,
-    DhPermissionRequiredDirective,
     DhUserAuditLogsComponent,
     DhUserMasterDataComponent,
-    DhUserRolesComponent,
+    DhPermissionRequiredDirective,
   ],
 })
-export class DhUserDrawerComponent {
-  private modalService = inject(WattModalService);
+export class DhUserDetailsComponent {
   private transloco = inject(TranslocoService);
   private toastService = inject(WattToastService);
+  private navigation = inject(DhNavigationService);
 
   drawer = viewChild.required<WattDrawerComponent>(WattDrawerComponent);
 
-  deactivateConfirmationModal = viewChild.required<WattModalComponent>(
-    'deactivateConfirmationModal'
-  );
+  // Router param
+  id = input.required<string>();
 
-  reActivateConfirmationModal = viewChild.required<WattModalComponent>(
-    'reActivateConfirmationModal'
-  );
-
-  closed = output<void>();
-
-  userId = signal<string | null>(null);
-  userIdWithDefaultValue = computed(() => this.userId() ?? '');
-
-  selectedUserQuery = lazyQuery(GetUserByIdDocument, {
-    fetchPolicy: 'no-cache',
-    nextFetchPolicy: 'no-cache',
-  });
+  selectedUserQuery = lazyQuery(GetUserDetailsDocument);
 
   selectedUser = computed(() => this.selectedUserQuery.data()?.userById);
   isLoading = computed(() => this.selectedUserQuery.loading());
 
   UserStatus = UserStatus;
 
   reInviteUserMutation = mutation(ReInviteUserDocument, {
-    refetchQueries: [UserOverviewSearchDocument],
-  });
-  reset2faMutation = mutation(Reset2faDocument, { refetchQueries: [UserOverviewSearchDocument] });
-  deactivateUserMutation = mutation(DeactivateUserDocument, {
-    refetchQueries: [UserOverviewSearchDocument],
+    refetchQueries: [GetUsersDocument, GetUserAuditLogsDocument],
   });
-  reActivateUserMutation = mutation(ReActivateUserDocument, {
-    refetchQueries: [UserOverviewSearchDocument],
+
+  reset2faMutation = mutation(Reset2faDocument, {
+    refetchQueries: [GetUsersDocument, GetUserAuditLogsDocument],
   });
 
   isReinviting = this.reInviteUserMutation.loading;
-  isDeactivating = this.deactivateUserMutation.loading;
-  isReActivating = this.reActivateUserMutation.loading;
 
   constructor() {
     effect(() => {
-      const userId = this.userId();
-      if (!userId) return;
-      this.selectedUserQuery.refetch({ id: userId });
+      const id = this.id();
+      const drawer = this.drawer();
+      if (!id || !drawer) return;
+      this.drawer().open();
+      this.selectedUserQuery.refetch({ id });
+      drawer.open();
```

Comment:
issue: the `drawer.open()` is called twice.

Reply:
Nice find!